File: ProjC1/src/projc1/Game/Coin.java

package projc1.Game;

class Coin {
    private Position position; // COMPOSICIÓN: Coin tiene Position
    private java.awt.Color color;
    private int value;
    private static final int SIZE = 20;
    
    public Coin(int x, int y) {
        this.position = new Position(x, y); // COMPOSICIÓN
        this.color = java.awt.Color.YELLOW;
        this.value = 10;
    }
    
    // ASOCIACIÓN: Usa RenderEngine para dibujarse
    public void draw(java.awt.Graphics2D g2d, RenderEngine renderer) {
        int x = position.getX();
        int y = position.getY();
        
        // Dibujar la moneda
        renderer.drawOval(g2d, x, y, SIZE, SIZE, color);
        renderer.drawOval(g2d, x, y, SIZE, SIZE, color.darker());
        
        // Punto brillante
        renderer.drawOval(g2d, x + 5, y + 5, 6, 6, java.awt.Color.WHITE);
        
        // Símbolo de moneda
        renderer.drawText(g2d, "$", x + 6, y + 14, 
                         new java.awt.Font("Arial", java.awt.Font.BOLD, 12), 
                         java.awt.Color.BLACK);
    }
    
    public boolean intersects(Position playerPos, int playerSize) {
        int x = position.getX();
        int y = position.getY();
        int px = playerPos.getX();
        int py = playerPos.getY();
        
        return px < x + SIZE && px + playerSize > x && 
               py < y + SIZE && py + playerSize > y;
    }
    
    public int getValue() { return value; }
    public Position getPosition() { return position; }
}

File: ProcC1_pyver/game/entities/coin.py

from game.components.position import Position


class Coin:
    SIZE = 20
    
    def __init__(self, x, y):
        self.position = Position(x, y)  # Composición
        self.color = "gold"
        self.value = 10
    
    def draw(self, canvas, renderer):
        x = self.position.get_x()
        y = self.position.get_y()
        
        # Moneda
        renderer.draw_oval(canvas, x, y, self.SIZE, self.SIZE, self.color)
        renderer.draw_oval(canvas, x, y, self.SIZE, self.SIZE, "#daa520")  # borde
        
        # Punto brillante
        renderer.draw_oval(canvas, x + 5, y + 5, 6, 6, "white")
        
        # Símbolo de moneda
        renderer.draw_text(canvas, "$", x + 8, y + 14, ("Arial", 10, "bold"), "black")
    
    def intersects(self, player_pos, player_size):
        x = self.position.get_x()
        y = self.position.get_y()
        px = player_pos.get_x()
        py = player_pos.get_y()
        
        return (px < x + self.SIZE and px + player_size > x and 
                py < y + self.SIZE and py + player_size > y)
    
    def get_value(self): return self.value
    def get_position(self): return self.position